### PR TITLE
Validate gas token balance when scheduling a payment

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-5231

https://user-images.githubusercontent.com/12637010/223320804-5696673c-3bd5-451b-9c68-58f9e95cc8c9.mov

### CODE REVIEW

File: packages/cardpay-sdk/sdk/scheduled-payment-module.ts [hunk 1]
```diff
@@ -1089,6 +1089,48 @@ export default class ScheduledPaymentModule {
     await waitUntilSchedulePaymentTransactionMined(hubRootUrl, scheduledPaymentId, authToken, options.listener);
   }
 
+  async estimateGasTokenToSchedule(
```
Comment (jurgenwerk) [inline packages/cardpay-sdk/sdk/scheduled-payment-module.ts:1092]: Could `estimateSchedulePaymentInGasToken` be a bit more descriptive descriptive name?

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts [hunk 2]
```diff
@@ -376,6 +383,53 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
     return state;
   });
 
+  @use requiredGasTokenToSchedulePayment = resource(() => {
+    const state: RequiredGasTokenToSchedulePaymentState = new TrackedObject({
+      isLoading: false,
+      isIndeterminate: false
```
Comment (jurgenwerk) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:389]: Can you explain how `isIndeterminate` is used and why is it needed? What is it that we can't determine?

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts [hunk 3]
```diff
@@ -376,6 +383,53 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
...
+    });
+    
+    let params = this.composeSchedulePaymentParams();
+    if (!this.isValid || !params) {
+      state.isIndeterminate = true;
```
Comment (jurgenwerk) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts:394]: Is `isIndeterminate`'s value used or read somewhere? I can't find that

File: packages/cardpay-sdk/sdk/scheduled-payment-module.ts (hunk 1 above)
Comment (FadhlanR) [reply packages/cardpay-sdk/sdk/scheduled-payment-module.ts]: updated

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts (hunk 2 above)
Comment (FadhlanR) [reply packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts]: Yep we don't need this field. Removed

File: packages/safe-tools-client/tests/acceptance/scheduling-test.ts [hunk 4]
```diff
@@ -407,6 +436,8 @@ module('Acceptance | scheduling', function (hooks) {
           'Calculating payment hash'
         )
       );
+
+      //scheduledPaymentCreateSpHashDelay = scheduledPaymentCreateSpHashDeferred?.promise;
```
Comment (lukemelia) [inline packages/safe-tools-client/tests/acceptance/scheduling-test.ts:440]: Should this comment be removed?

**Review by lukemelia — APPROVED** [review]

**Pull request merged**